### PR TITLE
SOW-24: notification banner

### PR DESCRIPTION
**Pull request by vferraro-scottlogic**
## Changes
- add notification banner to gds-components
- add banner to Export page
- [x] check target branch is main (should be changed automatically)

### CODE REVIEW

File: src/routes/Page.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export const Page = () => (
     <Header homeURL={PageRoute.Base} />
     <FormProvider>
       <div className="govuk-width-container app-width-container">
-        <main className="govuk-main-wrapper">
+        <main className="govuk-main-wrapper govuk-!-padding-top-0">
```
Comment (vferraro-scottlogic) [inline src/routes/Page.tsx:14]: unrelated to the ticket: page had too much padding on top

File: src/pages/create-timetable-page/CreateTimetablePage.tsx [hunk 2]
```diff
@@ -16,14 +16,22 @@ export const CreateTimetablePage = () => {
...
+    setStatusHasChanged,
   } = useFormContext();
 
   useEffect(() => {
     setTimetableEvents(DEFAULT_TIMETABLE_EVENTS);
     setDevelopmentPlan(DEFAULT_DEVELOPMENT_PLAN);
     setLoadedTimetableEvents(null);
     setLoadedDevelopmentPlan(null);
-  }, [setDevelopmentPlan, setLoadedDevelopmentPlan, setLoadedTimetableEvents, setTimetableEvents]);
+    setStatusHasChanged(null);
```
Comment (vferraro-scottlogic) [inline src/pages/create-timetable-page/CreateTimetablePage.tsx:27]: this part of the state wasn't being reset

File: lib/constants.ts [hunk 3]
```diff
@@ -19,19 +19,20 @@ export enum TimetableEventKey {
...
-  | TimetableEventKey.PlanWithdrawn
-  | TimetableEventKey.PlanPaused
-  | TimetableEventKey.PlanNotAdopted
->;
+export const statusChangeEvents = [
+  TimetableEventKey.PlanFoundUnsound,
+  TimetableEventKey.PlanWithdrawn,
+  TimetableEventKey.PlanPaused,
+  TimetableEventKey.PlanNotAdopted,
+] as const;
```
Comment (Ro4052) [inline lib/constants.ts:27]: Could we use this to spread into the set we use for filtering the default events below?

File: lib/gds-components/notification-banner/NotificationBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/**
+ * GDS component: https://design-system.service.gov.uk/components/notification-banner/
+ */
+
+import { toDefaultLocalDateString } from "../..//utils/timetable";
```
Comment (Ro4052) [inline lib/gds-components/notification-banner/NotificationBanner.tsx:5]: Looks like an extra `/` in the path (before "utils")

File: lib/gds-components/notification-banner/NotificationBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        <h2 className="govuk-notification-banner__title">Important</h2>
+      </div>
+      <div className="govuk-notification-banner__content">
+        <h3 className="govuk-notification-banner__heading">{title}</h3>
+        {date && (
+          <p className="govuk-body govuk-!-margin-bottom-1">
+            Date: {toDefaultLocalDateString(date)}
+          </p>
+        )}
+        <p className="govuk-body">{message}</p>
```
Comment (Ro4052) [inline lib/gds-components/notification-banner/NotificationBanner.tsx:34]: If `message` is also optional, should we do the same as we're doing for the date above, i.e. `{message && <p....`?

File: lib/constants.ts (hunk 3 above)
Comment (vferraro-scottlogic) [reply lib/constants.ts:27]: done :)

File: lib/gds-components/notification-banner/NotificationBanner.tsx (hunk 4 above)
Comment (vferraro-scottlogic) [reply lib/gds-components/notification-banner/NotificationBanner.tsx]: done :)

File: lib/gds-components/notification-banner/NotificationBanner.tsx (hunk 5 above)
Comment (vferraro-scottlogic) [reply lib/gds-components/notification-banner/NotificationBanner.tsx]: done :)

**Review by Ro4052 — DISMISSED** [review]

**Review by Ro4052 — APPROVED** [review]

**Pull request merged**